### ISSUE TITLE
autocomplete doesn't work for incomplete macros in lines with more marcos and/or brackets

### ISSUE THREAD

**Issue by danhaywood** [asciidoctor/asciidoctor-intellij-plugin#526]
Sometimes I find the plugin autocomplete for xref's fails to find anything at all.  I haven't noticed any errors in the error log, and if I type in the xref manually, then it sometimes pick up ok.

My guess is that there's a cache that's stale, or some other defect.  

As a user, having the ability to force a reload of caches (but without being as drastic as the File > Invalidate Caches route) would be nice.  Else some better feedback in the logs (but it's hard to know what to ask for here...)

**Comment by ahus1** [comment]
There might be multiple reasons for that: Maybe a missing increment of the lexer/parser versions so AsciiDoc missed to rebuild the indexes during update. Maybe asymmetric code for autocomplete and find-references.

Does IntelliJ rebuild the indexes after uninstall/re-install of the plugin?

Please post the version of the IDE and the plugin you are using. A small recording with a tool like [screentogif](https://www.screentogif.com/) taken in a publically accessible repo would be helpful.

**Comment by danhaywood** [comment]
I'll do that, but in the meantime, some more precise info: that if I type
"xref:" then xref isn't colour coded so presumably isn't therefore
triggering the intelligence logic.

As a workaround, if I type "xref:[]" then that does seem to cause xref to
become colour coded, and I can go back and use ctrl+space to get the
intelligence options.

Perhaps that extra info might help track down the issue.

**Comment by ahus1** [comment]
Thanks Dan, this helped indeed to track down the issue. 

I found a situation like the following, where an incomplete "xref:" followed by the caret followed by an opening bracket in the same line will not work with autocomplete: 

```
xref:<caret> and other xref:complete[] normal text
``` 

This will be part of an upcoming release. What version of the plugin/IntelliJ do you use (there are currently two active branches, 0.30.x for 2019.3.1+ and 0.31.x for 2020.2+)?

**Comment by danhaywood** [comment]
Still on 2020.1

**Comment by danhaywood** [comment]
which I think is the later version of the plugin

**Comment by ahus1** [comment]
Thanks, just wanted to know for what version to prepare a pre-release.

**Comment by ahus1** [comment]
There's now a pre-release  0.30.86 for 2019.3.1+ and pre-release 0.31.16 for 2020.2+

Please give it a try if it fixes your problem. 

The pre-release of the plugin is available from [GitHub releases](https://github.com/asciidoctor/asciidoctor-intellij-plugin/releases) and the [IntelliJ AsciiDoc EAP repository](https://intellij-asciidoc-plugin.ahus1.de/docs/contributors-guide/user/add-eap-repository-to-ide.html).

I close this issue as I was able to reproduce the issue and the solution locally. If the issue persists, please comment and I'll re-open the issue. 

Thanks for reporting this!

**Comment by ahus1** [comment]
I found another situation when auto-completion didn't work: sometimes IntelliJ didn't add a blank at the end of the line to its auto-completion magic string. This is now fixed in pre-release 0.31.19. 

The pre-release of the plugin is available from [GitHub releases](https://github.com/asciidoctor/asciidoctor-intellij-plugin/releases) and the [IntelliJ AsciiDoc EAP repository](https://intellij-asciidoc-plugin.ahus1.de/docs/contributors-guide/user/add-eap-repository-to-ide.html).
